ggml.c

```diff
@@ -16695,8 +16796,10 @@ struct ggml_cplan ggml_graph_plan(const struct ggml_cgraph * cgraph, int n_threa
 #if defined(GGML_USE_ACCELERATE) || defined(GGML_USE_OPENBLAS)
                     if (ggml_compute_forward_mul_mat_use_blas(node)) {
                         if (node->src[0]->type != GGML_TYPE_F32) {
-                            // here we need memory just for single 2D matrix from src0
-                            cur = ggml_type_size(GGML_TYPE_F32)*(node->src[0]->ne[0]*node->src[0]->ne[1]);
+                            // here we need memory for fully dequantized matrix from src0
+                            cur = ggml_type_size(GGML_TYPE_F32)
+                                * node->src[0]->ne[0]*node->src[0]->ne[1]
+                                * node->src[1]->ne[2]*node->src[1]->ne[3];
```

Comment:
Can be simplified to:

```suggestion
                            cur = ggml_type_size(GGML_TYPE_F32)*ggml_nelements(node->src[0]);
```

Reply:
Nevermind - it has to be like this to take into account the broadcast

Reply:
@ggerganov it looks like this suggestion was merged, but it sounds like you didn't mean for it to be?

Reply:
Fixed here: https://github.com/ggerganov/llama.cpp/pull/5094